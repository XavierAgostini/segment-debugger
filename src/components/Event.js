import React from 'react'
import { CheckCircleIcon } from 'evergreen-ui'
import moment from 'moment'
import CircleCheck from './CircleCheck'
import './Event.module.css'

const getEventName = (event) => {
  if (event.type === 'track') {
    return event.event
  }
  if (event.type === 'page' || event.type === 'screen') {
    if (event.properties.category && event.properties.name) {
      return event.properties.category + ' ' + event.properties.name
    } else if (event.properties.category) {
      return event.properties.category
    } else if (event.properties.name) {
      return event.properties.name
    }
    return event.properties.path
  }
  if (event.type === 'identify') {
    if (event.userId) {
      if (event.traits.email) {
         return `${event.userId} (${event.traits.email})`
      } else if (event.traits.name) {
        return `${event.userId} (${event.traits.name})`
      }
    } else {
      return `Anonymous Id: ${event.anonymousId}`
    }
  }
  if (event.type === 'group') return event.groupId
}

export default class Event extends React.Component {
  buttonClicked = () => {
    this.props.handleEventSelected(this.props.event)
  }

  render() {
    return (
      <button className="eventItem" onClick={this.buttonClicked} data-is-selected={this.props.isSelected}>
        <CircleCheck/>
        <div className="eventType">
          {this.props.event.type}
        </div>
        <div className="eventName">
          <div className="eventNameText">
            {getEventName(this.props.event)}
          </div>
        </div>
        <div className="eventTimeStamp">
          {moment(this.props.event.receivedAt).format('YYYY/MM/DD HH:mm:ss')}
        </div>
      </button>
    )
  }
}