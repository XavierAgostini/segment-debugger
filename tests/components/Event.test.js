import React from 'react';
import { shallow } from 'enzyme';
import Event from '../../src/components/Event'
import testEvents from '../seed/seed'

let wrapper, testEvent, handleEventSelected, isSelected

beforeEach(() => {
  testEvent = testEvents[0]
  handleEventSelected = jest.fn()
  isSelected = false
  wrapper = shallow(<Event event={testEvent} handleEventSelected={handleEventSelected} isSelected={isSelected}/>)
})

describe('<Event/>', () => {
  test('should render Event correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('div.eventNameText').text()).toEqual('Notification Shown')
    expect(wrapper.find('div.eventType').text()).toEqual('track')
  })
  test('should handle handleEventSelected', () => {
    wrapper.simulate('click')
    expect(handleEventSelected).toHaveBeenCalledWith(testEvent)
  })
})