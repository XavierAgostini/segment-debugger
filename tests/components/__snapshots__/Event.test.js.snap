// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event/> should render Event correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    event={
      Object {
        "anonymousId": "e1dbd770-1a96-48da-8bb9-2a294a11a571",
        "context": Object {
          "ip": "149.109.19.117",
          "library": Object {
            "name": "analytics.js",
            "version": "3.0.0",
          },
        },
        "event": "Notification Shown",
        "integrations": Object {},
        "messageId": "ajs-ac65f2dc-a7ba-496c-be3a-74a0de177c7f",
        "receivedAt": "2018-09-18T20:01:40.790Z",
        "sentAt": 1537300895790,
        "type": "track",
        "userId": "6a8e7fe9-09e8-480b-bf12-27c052035ebf",
      }
    }
    handleEventSelected={[MockFunction]}
    isSelected={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <div
          className="eventType"
        >
          track
        </div>,
        <div
          className="eventName"
        >
          <div
            className="eventNameText"
          >
            Notification Shown
          </div>
        </div>,
        <div
          className="eventTimeStamp"
        >
          2018/09/18 13:01:40
        </div>,
      ],
      "className": "eventItem",
      "data-is-selected": false,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "track",
          "className": "eventType",
        },
        "ref": null,
        "rendered": "track",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="eventNameText"
          >
            Notification Shown
          </div>,
          "className": "eventName",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Notification Shown",
            "className": "eventNameText",
          },
          "ref": null,
          "rendered": "Notification Shown",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018/09/18 13:01:40",
          "className": "eventTimeStamp",
        },
        "ref": null,
        "rendered": "2018/09/18 13:01:40",
        "type": "div",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <div
            className="eventType"
          >
            track
          </div>,
          <div
            className="eventName"
          >
            <div
              className="eventNameText"
            >
              Notification Shown
            </div>
          </div>,
          <div
            className="eventTimeStamp"
          >
            2018/09/18 13:01:40
          </div>,
        ],
        "className": "eventItem",
        "data-is-selected": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "track",
            "className": "eventType",
          },
          "ref": null,
          "rendered": "track",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="eventNameText"
            >
              Notification Shown
            </div>,
            "className": "eventName",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Notification Shown",
              "className": "eventNameText",
            },
            "ref": null,
            "rendered": "Notification Shown",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018/09/18 13:01:40",
            "className": "eventTimeStamp",
          },
          "ref": null,
          "rendered": "2018/09/18 13:01:40",
          "type": "div",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
